src/types.ts

```diff
@@ -0,0 +1,108 @@
+import type { JITI } from "jiti";
+import type { JITIOptions } from "jiti/dist/types";
+import type { DotenvOptions } from "./dotenv";
+
+export interface ConfigLayerMeta {
+  name?: string;
+  [key: string]: any;
+}
+
+export type UserInputConfig = Record<string, any>;
+
+export interface C12InputConfig<
+  T extends UserInputConfig = UserInputConfig,
+  MT extends ConfigLayerMeta = ConfigLayerMeta
+> {
+  $test?: T;
+  $development?: T;
+  $production?: T;
```

Comment:
would it make sense for these keys to be customisable? We could do it with a third generic above, if so.

Reply:
First generic for `InputConfig` allows to extend more meta but ideally we might keep `$` prefix for built-in features to avoid key conflicts when we introduce more.

Env aliases are also built-in feature. I intentionally typed 3 because we might add a check to only support 3 options (while we could add more built-ins later if you think worth...)

Reply:
I mean we could have something like:

`Env extends string = 'test' | 'development' | 'production'`

That would also allow _removing_ the `$test`, `$development` keys if not appropriate for the project using c12.

Reply:
We could make it opt-in but it makes usage harder across projects. Do you think there is a reason we don't want to support top-level env aliases? (FYI currently we allow [all env aliases](https://github.com/unjs/c12/blob/9df04984e53ef8e4ceef3fb5cc2453007a6708ee/src/loader.ts#L339) while only typing well-known `NODE_ENV` values)

Reply:
(resolved in discord discussion)